src/test/unit/test_tracer.py

```diff
@@ -468,6 +472,23 @@ def lambda_test_function(event, context):
     assert function_span["error"]["message"] == expected_message
 
 
+def test_not_jsonable_return_value_non_python37(monkeypatch, context, caplog):
+    monkeypatch.setenv("AWS_EXECUTION_ENV", "AWS_Lambda_python3.8")
+
+    @lumigo_tracer()
+    def lambda_test_function(event, context):
+        return {"a": datetime.datetime.now()}
+
+    lambda_test_function({}, context)
+
+    function_span = SpansContainer.get_span().function_span
+    assert function_span["return_value"] is None
+    assert "error" not in function_span
+    log = caplog.records[-1]
+    assert log.levelno == logging.ERROR
+    assert "Could not serialize the return value of the lambda" in log.message
```

Comment:
IMO it's better to do something like this because then we don't care about the order of the logs.
```suggestion
    next(log for log in caplog.records if log.levelno == logging.ERROR and "Could not serialize the return value of the lambda" in log.message)
```